src/io/flutter/run/SdkRunConfig.java

```diff
@@ -84,11 +98,56 @@ public LaunchState getState(@NotNull Executor executor, @NotNull ExecutionEnviro
     final Project project = env.getProject();
     final RunMode mode = RunMode.fromEnv(env);
     final Module module = ModuleUtil.findModuleForFile(mainFile.getFile(), env.getProject());
-
     final LaunchState.Callback callback = (device) -> {
       if (device == null) return null;
 
-      final GeneralCommandLine command = fields.createFlutterSdkRunCommand(project, device, mode);
+      final GeneralCommandLine command = getCommand(env, device);
+      {
+        // Workaround for https://github.com/flutter/flutter/issues/16766
+        // TODO(jacobr): remove once flutter tool incremental building works
+        // properly with --track-widget-creation.
+        final Path buildPath = command.getWorkDirectory().toPath().resolve("build");
+        final Path cachedParametersPath = buildPath.resolve("last_build_run.json");
+        final String[] parametersToTrack = {"--preview-dart-2", "--track-widget-creation"};
+        final JsonArray jsonArray = new JsonArray();
```

Comment:
If we wanted to simplify this, we could just put all the launch args into a text file, do a string compare on the current args, and delete the stamp file if there were differences.

Reply:
(looks good either way)

Reply:
The problem with using all the args is then we get spurious deletes if you just change the device id related args this way we don't do invasive things to the build unless you change the --track-widget-creation or --preview-dart-2 flags.
Technically we only really need to use --track-widget-creation but I'm being a bit paranoid.